workflow-templates/python-cross-platform-support.yml

```diff
@@ -77,9 +77,9 @@ jobs:
         uses: actions/checkout@v3
 
       - name: Download Secure Install
-        uses: actions/download-artifact@v3
+        uses: actions/download-artifact@v4
         with:
-          name: secure_install_${{ matrix.os }}
+          name: secure_install_${{ matrix.os }}_${{steps.package-version.outputs.VERSION_STRING}}
```

Comment:
This is in a separate job, so either need to add the additional step of generating the version string. Or, and really preferred, make a stand alone job of getting the version string that can be referenced in other jobs and their steps.